plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -263,6 +271,21 @@ private JdbcTableHandle flushAttributesAsQuery(ConnectorSession session, JdbcTab
         ImmutableList.Builder<ConnectorExpression> projections = ImmutableList.builder();
         ImmutableList.Builder<Assignment> resultAssignments = ImmutableList.builder();
         ImmutableMap.Builder<String, String> expressions = ImmutableMap.builder();
+
+        List<List<JdbcColumnHandle>> groupingSetsAsJdbcColumnHandles = groupingSets.stream()
+                .map(groupingSet -> groupingSet.stream()
+                        .map(JdbcColumnHandle.class::cast)
+                        .collect(toImmutableList()))
+                .collect(toImmutableList());
+        Optional<List<JdbcColumnHandle>> tableColumns = handle.getColumns();
+        groupingSetsAsJdbcColumnHandles.stream()
+                .flatMap(List::stream)
+                .distinct()
+                .peek(groupKey -> verify(
+                        tableColumns.map(columns -> columns.contains(groupKey)).orElse(true),
+                        "applyAggregation called with a grouping column which was not included in the table handle"))
+                .forEach(newColumns::add);
```

Comment:
What query is sent to eg PostgreSQL when you run `SELECT count(*) FROM nation GROUP BY region` in Trino?

are we reading `region` unnecessarily or not?

worth adding a comment at https://github.com/trinodb/trino/blob/825e6d82cfd3dee98038f14b38a55e0d1798ae35/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java#L312

Reply:
`SELECT "regionkey", count(*) AS "_pfgnrtd_0" FROM "tpch"."nation" GROUP BY "regionkey"`

We are

Reply:
we should follow up on this, since it means we're reading more data than necessary  for some queries

Reply:
https://github.com/trinodb/trino/issues/9021